base/acme/src/main/java/org/dogtagpki/acme/database/PostgreSQLDatabase.java

```diff
@@ -413,6 +413,47 @@ public ACMEOrder getOrder(String orderID) throws Exception {
         return orders;
     }
 
+    public ACMEOrder getOrderByCertificate(String certID) throws Exception {
+
+        logger.info("Getting order for certificate " + certID);
+
+        String sql = statements.getProperty("getOrderByCertificate");
+        logger.info("SQL: " + sql);
+
+        ACMEOrder order = new ACMEOrder();
+
+        try (PreparedStatement ps = connection.prepareStatement(sql)) {
+            ps.setString(1, certID);
+
+            try (ResultSet rs = ps.executeQuery()) {
+
+                if (!rs.next()) {
+                    return null;
+                }
+
+                order.setID(rs.getString("id"));
+                order.setAccountID(rs.getString("account_id"));
+                order.setStatus(rs.getString("status"));
+
+                Timestamp expires = rs.getTimestamp("expires");
+                order.setExpirationTime(new Date(expires.getTime()));
+
+                Timestamp notBefore = rs.getTimestamp("not_before");
+                order.setNotBeforeTime(notBefore == null ? null : new Date(notBefore.getTime()));
+
+                Timestamp notAfter = rs.getTimestamp("not_after");
+                order.setNotAfterTime(notAfter == null ? null : new Date(notAfter.getTime()));
+
+                order.setCertID(certID);
```

Comment:
So this function returns the last order with this certificate, whereas the in-memory one returns the first one matching. More than one order for the same cert happens when the cert is renewed or the request failed for some reason, right? So do we want to standardize our selection here somehow? Only open ones? Most recently filled?

Reply:
I think we can only have one order per certificate. If the certificate is renewed it will be handled like a new request, i.e. it will create a new order which generates new certificate. If the request failed, no certificate will be issued, and the order will be marked invalid.

Since there's only one order per certificate, the first and last orders are the same. Also, the `orders` map in the `InMemoryDatabase` and the `orders` table in `PostgreSQLDatabase` are not sorted so there isn't really a first or last order.

Reply:
I agree @edewata; the implementations seem fine.  But please add some clarifying comments.

Reply:
I added a Javadoc comment in the base class for this method.

Reply:
Oh oh oh. Sorry, I realized what my issue was now. 

```java
try (ResultSet rs = ps.executeQuery()) {
```

For some reason, I misread that line as a loop over all returned data. My comment doesn't make sense.